vcd/resource_vcd_network_routed.go

```diff
@@ -263,10 +258,16 @@ func resourceVcdNetworkRead(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
+func resourceVcdNetworkDeleteLocked(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	return resourceVcdNetworkDelete(d, meta)
```

Comment:
Just to be on a safe side, I double checked that the order of defer execution is as we are expecting even with a function in return statement, and it is :) Small illustration:

```
package main

import "fmt"

func resourceVcdNetworkDeleteLocked() error {
    fmt.Println("lock")
	defer fmt.Println("unlock")

    return resourceVcdNetworkDelete()
}

func resourceVcdNetworkDelete() error {
	fmt.Println("delete")

	return nil
}

func main() {
	resourceVcdNetworkDeleteLocked()
}
```

Output:
```
lock
delete
unlock
```

Reply:
Yes, I confirmed that too :) before using